report/metadata_template.go

```diff
@@ -124,11 +124,14 @@ func (e MetadataTemplates) Copy() MetadataTemplates {
 // Merge merges two sets of MetadataTemplates so far just ignores based
 // on duplicate id key
 func (e MetadataTemplates) Merge(other MetadataTemplates) MetadataTemplates {
-	result := e.Copy()
+	if e == nil && other == nil {
+		return nil
+	}
+	result := make(MetadataTemplates, len(e))
```

Comment:
I don't think + would be a good idea - won't the keys be the same or nearly the same in most cases? As for max... if only golang didn't make me write my own function for that. Also this mini optimisation would apply in more places than the ones I changed. So I'd rather leave it for now.